tests/test_profiler.py

```diff
@@ -0,0 +1,107 @@
+from wandb.sdk import internal
+import time
+import os
+import wandb
+import pytest
+import sys
+from wandb.sdk.internal import profiler
+
+from wandb.sdk.internal.profiler import ProfilerWatcher
+from wandb.proto import wandb_internal_pb2 as pb
+
+
+@pytest.mark.skipif(
+    sys.version_info >= (3, 9), reason="PyTorch not currently stable for 3.9"
+)
+def test_profiler_watcher_v1(
+    runner, mock_server, test_settings, backend_interface, internal_hm, mocked_run
```

Comment:
I think this would be cleaner to use `publish_util` fixture. we can chat about what that would look like.  it should simplify things.